.github/workflows/build_and_deploy_ext2_img.yml

```diff
@@ -0,0 +1,199 @@
+              echo "::error title=Invalid Dockerfile path::No file found at ${{ github.event.inputs.DOCKERFILE_PATH }}"
+              gh run cancel ${{ github.run_id }}
+              gh run watch ${{ github.run_id }}
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }} # As required by the GitHub-CLI
+
+      # Create directory to host the image from.
+      - run: sudo mkdir -p $DEPLOY_DIR
+
+      # Move required files for gh-pages deployment to the deployment directory $DEPLOY_DIR.
+      - run: sudo mv assets examples tun xterm favicon.ico index.html login.html network.js scrollbar.css serviceWorker.js tower.ico $DEPLOY_DIR
+
+      # Set the docker build context and Dockerfile path env vars.
+      - run: |
+         echo "DOCKER_BUILD_CONTEXT=$(dirname ${{ github.event.inputs.DOCKERFILE_PATH }} )" >> $GITHUB_ENV
+         echo "DOCKERFILE_PATH=${{ github.event.inputs.DOCKERFILE_PATH }}" >> $GITHUB_ENV
+      
+      # Build the i386 Dockerfile image.
+      - run: docker build $DOCKER_BUILD_CONTEXT --tag $TAG --file $DOCKERFILE_PATH --platform=i386
+      
+      # Run the docker image so that we can export the container.
+      - run: docker run -d $TAG
+
+      # We create and mount the base ext2 image to extract the Docker container's filesystem its contents into.
+      - name: Create ext2 image.
+        run: |
+         # Preallocate space for the ext2 image
+         sudo fallocate -l $IMAGE_SIZE ${DEPLOY_DIR}/${IMAGE_NAME}
+         # Format to ext2 linux kernel revision 0
+         sudo mkfs.ext2 -r 0 ${DEPLOY_DIR}/${IMAGE_NAME}
+         # Mount the ext2 image to modify it
+         sudo mount -o loop -t ext2 ${DEPLOY_DIR}/${IMAGE_NAME} /mnt/
+      
+      # We choose to use Docker export instead of Docker save because we only care about the final result and not the metadata and separate layers of the Docker image.
+      - name: Export and unpack container filesystem contents into mounted ext2 image.
+        run: | 
+          docker export $(sudo docker ps -aq) | sudo tar -x -C /mnt/
+          sudo umount /mnt/
+      # Result is an ext2 image for webvm.
+
+      # This step adjusts the default index.html file by performing the following steps:
+      #   1. Replacing the 'create cx app' line to fit our ext2 URL.
+      #   2. Setting the uid and gid to the values supplied as input parameters by replacing the var opts line.
+      #   3. Replacing all occurrences of CX_VERSION with the value supplied as an input parameter.
+      - name: Adjust index.html
+        if: ${{ github.event.inputs.MANUAL_INDEX_HTML == 'false' }}
+        run: |
+          sudo sed -i 's/^\(\s*\)CheerpXApp.create(.*/\1CheerpXApp.create({devices:[{type:"bytes",url:"${{ env.IMAGE_NAME }}",name:"block1"}],mounts:[{type:"ext2",dev:"block1",path:"\/"},{type:"cheerpOS",dev:"\/app",path:"\/app"},{type:"cheerpOS",dev:"\/str",path:"\/data"}], networkInterface}).then(runTest, failCallback);/g' ${{ env.DEPLOY_DIR}}index.html
```

Comment:
Define variables (same as CX_VERSION) for all the parts that might be replaced. This will allow simplified regexps and will make the "manual_index" option redundant since people can still force other paths by just writing the intended value instead of using the variable.

For completeness, when the variables are defined in index.html, add a comment that says that the variables are replaced by GitHub actions